src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -888,6 +888,16 @@ private void EnsureSessionHandleExists(WinHttpRequestState state)
 
                 HttpResponseMessage responseMessage = WinHttpResponseParser.CreateResponseMessage(state, _doManualDecompressionCheck);
                 state.Tcs.TrySetResult(responseMessage);
+
+                if((responseMessage.StatusCode == HttpStatusCode.MultipleChoices ||
+                    responseMessage.StatusCode == HttpStatusCode.MovedPermanently ||
+                    responseMessage.StatusCode == HttpStatusCode.Redirect ||
+                    responseMessage.StatusCode == HttpStatusCode.RedirectMethod ||
+                    responseMessage.StatusCode == HttpStatusCode.RedirectKeepVerb) &&
+                   state.RequestMessage.RequestUri.Scheme == Uri.UriSchemeHttps &&
+                   responseMessage.Headers.Location?.Scheme == Uri.UriSchemeHttp) {
```

Comment:
nit: bracing style should match rest of file, i.e. Allman-style

Reply:
Are the urls in the payload or should they be included in the trace message?

Reply:
They're not in the payload. so I'll add it to the message. What do you think of wording it something like this?
```
$"Insecure https to http redirect from {RequestUri} to {DestinationUri} blocked."
```